[DOCS] Rework conceptual info for ILM. (#52181)
elastic/elasticsearch: docs/reference/ilm/set-up-lifecycle-policy.asciidoc

[role="xpack"]
[testenv="basic"]
[[set-up-lifecycle-policy]]
== Configure lifecycle policy [[ilm-policy-definition]]

For {ilm-init} to manage an index, a valid policy 
must be specified in the `index.lifecycle.name` index setting. 

To configure a lifecycle policy for rolling indices, 
you create the policy and add it to the index template.

To use a policy to manage an index that doesn't roll over,
you can specify the policy directly when you create it.

[float]
[[ilm-create-policy]]
=== Create lifecycle policy

> You use the Create policy API to define a new lifecycle policy. 

Reviewer: ```suggestion
You use the <<ilm-put-lifecycle,create policy API>> to define a new lifecycle policy. 
```

> For example, the following request creates `my_policy`, a simple

Reviewer: ```suggestion
For example, the following request creates `my_policy`, a
```

policy that defines a hot and and delete phase.

> When the index gets to 25GB, it rolls over directly to the delete phase.

Reviewer: ```suggestion
When the index reaches o 25GB, it rolls over directly to the delete phase.
```

}
------------------------

<1> Roll over the index when it reaches 25GB in size
<2> Delete the index 30 days after rollover

NOTE: {ilm-init} policies are stored in the global cluster state and can be included in snapshots by  

> setting `include_global_state` parameter to `true`. 

Reviewer: > setting `include_global_state` parameter to `true`. 

It's not clear which API this parameter belongs to. I'd mention and link to them here.

<1> Set this initial index to be the write index for this alias.

Now you can start indexing data to the rollover alias specified in the lifecycle policy. 
With the sample `my_policy` policy, the rollover action is triggered once the initial
index exceeds 25GB. 

> {ilm} then creates a new index that becomes the write index for the `test-alias`.

Reviewer: ```suggestion
{ilm-init} then creates a new index that becomes the write index for the `test-alias`.
```

[float]
[[apply-policy-manually]]
=== Apply lifecycle policy manually

When you create an index directly, you can apply a lifecycle policy 
by specifying the `index.lifecycle.name` setting.

> This causes {ilm} to immediately start managing the index.

Reviewer: ```suggestion
This causes {ilm-init} to immediately start managing the index.
```
